optuna/storages/rdb/storage.py

```diff
@@ -602,3 +594,137 @@ def __del__(self):
+
+        version_info = models.VersionInfoModel(
+            schema_version=models.SCHEMA_VERSION, library_version=version.__version__)
+
+        session.add(version_info)
+        self._commit_with_integrity_check(session)
+
+    def _init_alembic(self):
+        # type: () -> None
+
+        logging.getLogger('alembic').setLevel(logging.WARN)
+
+        context = alembic.migration.MigrationContext.configure(self.engine.connect())
+        is_initialized = context.get_current_revision() is not None
+
+        if is_initialized:
+            # The `alembic_version` table already exists and is not empty.
+            return
+
+        if self._is_alembic_supported():
+            revision = self.get_head_version()
+        else:
+            # The storage has been created before alembic is introduced.
+            revision = self._get_base_version()
+
+        self._set_alembic_revision(revision)
+
+    def _set_alembic_revision(self, revision):
+        # type: (str) -> None
+
+        context = alembic.migration.MigrationContext.configure(self.engine.connect())
+        script = self._create_alembic_script()
+        context.stamp(script, revision)
+
+    def _check_table_schema_compatibility(self):
+        # type: () -> None
+
+        session = self.scoped_session()
+
+        # NOTE: After invocation of `_init_version_info_model` method,
+        #       it is ensured that a `VersionInfoModel` entry exists.
+        version_info = models.VersionInfoModel.find(session)
+        assert version_info is not None
+
+        current_version = self.get_current_version()
+        head_version = self.get_head_version()
+        if current_version == head_version:
+            return
+
+        message = 'The runtime optuna version {} is no longer compatible with the table schema ' \
+                  '(set up by optuna {}). '.format(version.__version__,
+                                                   version_info.library_version)
+        known_versions = self.get_all_versions()
+        if current_version in known_versions:
+            message += 'Please execute `$ optuna storage upgrade --storage $STORAGE_URL`' \
+                       'for upgrading the storage.'
+        else:
+            message += 'Please try updating optuna to the latest version by '\
+                       '`$ pip install -U optuna`.'
+
+        raise RuntimeError(message)
+
+    def _is_alembic_supported(self):
+        # type: () -> bool
+
+        session = self.scoped_session()
+
+        version_info = models.VersionInfoModel.find(session)
+        if version_info is None:
+            return True
+
+        return version_info.schema_version == models.SCHEMA_VERSION
+
+    def _create_alembic_script(self):
+        # type: () -> alembic.script.ScriptDirectory
+
+        config = _create_alembic_config(self.url)
+        script = alembic.script.ScriptDirectory.from_config(config)
+        return script
+
+
+def _create_alembic_config(url):
```

Comment:
I think `_create_alembic_config` can be a static method of `RDBStorage` like `_fill_storage_url_template` since it is only used in `RDBStorage`.